src/main/java/org/jenkinsci/plugins/workflow/cps/replay/ReplayAction.java

```diff
@@ -135,27 +151,38 @@ private ReplayAction(Run run) {
             return false;
         }
 
-        CpsFlowExecution exec = getExecution();
-        if (exec == null) {
-            return false;
+        CpsFlowExecution exec = getExecutionLazy();
+        if (exec != null) {
+            return exec.isSandbox() || Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER); // We have to check for ADMIN because un-sandboxed code can execute arbitrary on-master code
+        } else {
+            // If the execution hasn't been lazy-loaded then we will wait to do deeper checks until someone tries to lazy load
+            // OR until isReplayableSandboxTest is invoked b/c they actually try to replay the build
+            return true;
         }
-        if (exec.isSandbox()) {
+    }
+
+    /** Runs the extra tests for replayability beyond {@link #isEnabled()} that require a blocking load of the execution. */
+    /* accessible to Jelly */ public boolean isReplayableSandboxTest() {
+        CpsFlowExecution exec = getExecutionBlocking();
+        if (exec != null) {
+            if (!exec.isSandbox()) {
+                // We have to check for ADMIN because un-sandboxed code can execute arbitrary on-master code
+                return Jenkins.getActiveInstance().hasPermission(Jenkins.ADMINISTER);
```

Comment:
`Jenkins.RUN_SCRIPTS`?

Reply:
You're right circa https://github.com/jenkinsci/scriptler-plugin/pull/32/files - fixed